src/smefit/optimize/analytic.py

```diff
@@ -147,6 +149,28 @@ def run_sampling(self):
             @ diff_sm
         )
 
+        # Compute some metrics of the fit result
+        coeffs_name = sorted(self.coefficients.free_parameters.index)
+
+        fit_result["best_fit_point"] = dict(zip(coeffs_name, coeff_best))
+
+        # compute max log likelihood
+        chi2_tot = chi2.compute_chi2(
+            self.loaded_datasets,
+            coeff_best,
```

Comment:
The chi2 cannot be computed this way in case of constraints in the runcard, it throws. With the runcard attached I got the following error: `ValueError: Size of label 'j' for operand 1 (46) does not match previous terms (45). The solution is to do 

```
        self.coefficients.set_free_parameters(coeff_best)
        self.coefficients.set_constraints()

        # compute max log likelihood
        chi2_tot = chi2.compute_chi2(
            self.loaded_datasets,
            self.coefficients.value,
            self.use_quad,
            self.use_multiplicative_prescription,
        )
        max_logl = float(-0.5 * chi2_tot)
```

This issue doesn't show up when running ultranest as it already provides the maximum likelihood itself such that it does not need to be computed separately starting from the chi2 on the best fit point.

[LHC_NLO_LIN_GLOB_PR91.yaml.zip](https://github.com/user-attachments/files/17830076/LHC_NLO_LIN_GLOB_PR91.yaml.zip)

Reply:
Great catch, I fixed it :)